code/modules/mob/inventory.dm

```diff
@@ -212,3 +222,25 @@
 		if(slot_r_hand)
 			return r_hand
 	return null
+
+// Called when a mob moves from a spot where dropping is forbidden, to a spot
+// where dropping is allowed - so taking off your jumpsuit in a mech doesn't let
+// your ID stay on - or let you carry a large O2 tank in suit storage without
+// the hardsuit itself
+/mob/proc/update_item_drops()

```

Comment:
Yes, that's exactly what should happen. If your hand comes off, or your suit vaporizes for some reason, everything will remain in place until you leave the mech/pod/whatever, at which point this proc will update those slots and cause anything loose to fall off.